### PR TITLE
Optimize issue fetching

### PR DESCRIPTION
**Pull request by johnclary**
Fixes https://github.com/cityofaustin/atd-data-tech/issues/10465

### Steps to test:
[Deploy preview](https://deploy-preview-39--atd-product.netlify.app/)
Please test everything:
- product list
- projects list, filters, search, chart
- project issue details (all tabs)
- product issue details (all tabs)
- about page

Locally, run `npm run build` and verify that there are no lint errors.

### CODE REVIEW

File: components/shared/indexIssueDetails/Issues.js [hunk 1]
```diff
@@ -47,41 +18,33 @@ function getParentLabel(labels, type) {
...
   );
-}
 
-export default function Issues(props) {
-  const context = React.useContext(IssuesContext);
-  const indexType = props.indexType;
+  // if there is no parent label configured in github, we pass `null` to SWR to prevent it from fetching
+  const { error, isLoaded, data } = useSocrata({
+    url: parentLabel ? url : null,
+  });
```
Comment (johnclary) [inline components/shared/indexIssueDetails/Issues.js:32]: docs on conditional data fetching with swr:
https://swr.vercel.app/docs/conditional-fetching#conditional

File: components/utils.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+};
+
+export function useSocrata({ url }) {
+  const { data, error } = useSWR(url, fetcher);
+  return {
+    data: data || [],
+    isLoaded: !!error || !!data,
+    error: error,
+  };
+}
```
Comment (johnclary) [inline components/utils.js:26]: here's the new data fetcher. the rest of this file contains functions extracted from elsewhere.

File: components/utils.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      return <RiBankFill />;
+    case "Research":
+      return <FaSearch />;
+    case "Data":
+      return <FaDatabase />;
+    case "Documentation":
+      return <FaEdit />;
+    default:
+      return null;
+  }
```
Comment (johnclary) [inline components/utils.js:111]: if we want more icons to render on the "Issues" tab, add support for more issue types here.

File: components/wrappers/IssuesContextWrapper.js [hunk 4]
```diff
@@ -1,135 +1,62 @@
...
+  const dataHandled = useMemo(() => {
+    if (!data || data.length === 0) {
+      return [];
+    }
+    return handleIssueData(data);
   }, [data]);
 
+  const projectIssues = useProjectIssues(dataHandled);
+  const productIssues = useProductIssues(dataHandled);
+  const workgroups = useWorkgroups(dataHandled);
```
Comment (johnclary) [inline components/wrappers/IssuesContextWrapper.js:50]: these memoized hooks will only do their heavy lifting once 😅

**Comment by amenity** [comment]
So speedy!! 🔥 Everything opened a spinning load wheel anywhere. 💯

**Review by amenity — APPROVED** [review]

**Comment by roseeichelmann** [comment]
So I tested this both with the Netlify preview and locally, and I found a lot of instances where the issues tab for a project just loaded forever in both the Netlify preview and my local run, like so:

image

I tried refreshing the page for all of them and it would still load forever. I listed out all of the projects where this happened below.

**In progress:**
ATD executive dashboard
Private-Public Partnership (P3) Management
Banner Payments and Permits
AMANDA FY23 Fee Updates

**Completed:**
Bus Ridership Incentives Project (cancelled)
Bike Share Benefit
Parking Enforcement Dispatch Tracking
Transportation Data Lake
OSM Bicycle Data Update
Digitize Residential Permit Parking Petition
MDS Upgrades
E-Citation System for Parking Enforcement
SD23 Wave 1 Strategic Reporting
Development Review Workflow Management
Vision Zero Viewer v1.0
SD23 Wave 2 Strategic Reporting

**Backlog:**
Digitize Travel Requests and Reimbursements
DAPCZ Agenda Management
ROW Activity Dashboard
Open Data for ROW Permitting
Mobile Signs Work Orders
ATD Vehicle/Tag Checkout

**Needs scoping:**
Barricade Rental for Special Events
Integrated Parking Citation Reporting

There were also a few projects and products that had the following errors on the issues tab:

Screenshot from 2022-10-19 17-52-31
Screenshot from 2022-10-19 17-50-44
Screenshot from 2022-10-19 17-44-02
Screenshot from 2022-10-19 17-14-22
Screenshot from 2022-10-19 17-52-45

Other than that, the rest of the website worked and was very speedy in loading issues and pages! Locally I ran `npm run build` and got no linting errors.

**Comment by amenity** [comment]
Ooooh, nice find, Rose! I think this might actually be exposing loose ends on the Product end of things: All of these projects are missing a project-specific label, so there are actually no issue to load. I'll open an issue for individual PMs to update these.

**Comment by johnclary** [comment]
yes, in this PR i had changed the way we handle mis-labeled projects. i rolled back that change so that we see "No issues found" in those cases.

there was also a separate bug that surfaced b/c of they were encoding labels that that had an `&` symbol in them. this should be all set, but for fixing these merge conflicts.

**Pull request merged**